process_scheduling_algorithms/preemptive_priority_scheduling.c

```diff
@@ -0,0 +1,158 @@
+/**
+ * @file
+ * @brief
+ * [Preemptive Priority
+ * Scheduling]
+ *  Preemptive Priority Scheduling is a CPU scheduling algorithm
+    that assigns priorities to processes and selects the process
+    with the highest priority to execute. If a higher-priority
+    process becomes available, it preempts the currently executing
+    process. This algorithm ensures that higher-priority processes
+    are given preference, but it may lead to starvation of lower-priority
+    processes if not handled properly.*
```

Comment:
```suggestion
 *  Preemptive Priority Scheduling is a   CPU scheduling algorithm
 *  that assigns priorities to processes and selects the process
 *  with the highest priority to execute. If a higher-priority
 *  process becomes available, it preempts the currently executing.
 *  This algorithm ensures that higher-priority processes
 *  are given preference, but it may lead to starvation of lower-priority
 *   if not handled properly.*
```